go/offline/rpc_cache.go

```diff
@@ -121,40 +126,96 @@ func (c *RPCCache) put(mctx libkb.MetaContext, version Version, rpcName string,
 	return err
 }
 
-// Serve an RPC out of the offline cache. The machinery only kicks into gear if the `oa` OfflineAvailability mode is set
-// to BEST_EFFORT. If not, then just use the function `handler` which does the main work of handling the RPC. Note that
-// `handler` already has the argument "baked in" or "curried in", and also, is responsible for setting a result of the
-// right type in the caller's stack frame. `handler` also returns the return value for the RPC as an inteface, so it
-// can be inserted into the offline cache in the success case. We also pass this function a `version`, which will tell
-// the cache-access machinery to fail if the wrong version of the data is cached. Next, we pass the `rpcName`, the
-// argument, and the pointer to which the result is stored if we hit the cache. This flow is unfortunately complicated,
-// but the issue is that we're trying to maintain runtime type checking, and it's hard to do without generics.
-func (c *RPCCache) Serve(mctx libkb.MetaContext, oa keybase1.OfflineAvailability, version Version, rpcName string, encrypted bool, arg interface{}, res interface{},
-	handler func(mctx libkb.MetaContext) (interface{}, error)) (err error) {
+// Serve an RPC out of the offline cache. The machinery only kicks
+// into gear if the `oa` OfflineAvailability mode is set to
+// BEST_EFFORT. If not, then just use the function `handler` which
+// does the main work of handling the RPC. Note that `handler` must
+// not modify anything in the caller's stack frame; it might be run in
+// a background goroutine after this function returns, to populate the
+// cache. `handler` also returns the return value for the RPC as an
+// interface, so it can be inserted into the offline cache in the
+// success case. We also pass this function a `version`, which will
+// tell the cache-access machinery to fail if the wrong version of the
+// data is cached. Next, we pass the `rpcName`, the argument, and the
+// pointer to which the result is stored if we hit the cache. This
+// flow is unfortunately complicated, but the issue is that we're
+// trying to maintain runtime type checking, and it's hard to do
+// without generics.
```

Comment:
This comment is now out-of-date ^

Reply:
@maxtaco  I did my best to update the comment -- what about it is out of date now?  The last sentence still seems valid to me, and I updated all the stuff about what the handler should be doing.

Reply:
I was thinking the last sentence is now no longer needed since you're casting back out of the handle (rather than currying in the reply pointer).

Reply:
Hmm ok thanks.  I'll get rid of it.

Reply:
Not a big deal either way